### PR TITLE
UseItem

### PR DESCRIPTION
**Pull request by louis1706**
- instead of https://github.com/Exiled-Team/EXILED/pull/1352

### CODE REVIEW

**Comment by louis1706** [comment]
Need test for all the folowiing itemtype
ItemType.Scp500 (or any other consumable item)
ItemType.Scp330
ItemType.Scp244
ItemType.Scp268

File: Exiled.API/Features/Player.cs [hunk 1]
```diff
@@ -1912,6 +1912,21 @@ public void Heal(float amount, bool overrideMaxHealth = false)
                 Health += amount;
         }
 
+        /// <summary>
+        /// Used an item for giving effect of it.
```
Comment (Ficus-x) [inline Exiled.API/Features/Player.cs:1916]: Maybe "Uses an item for giving its effect."? I think the method name should be also Heal

Comment (louis1706) [reply Exiled.API/Features/Player.cs:1916]: no because it's can use other things who are not heal like Scp244

**Review by Jesus-QC — APPROVED** [review]

File: Exiled.API/Features/Player.cs [hunk 2]
```diff
@@ -1912,6 +1912,21 @@ public void Heal(float amount, bool overrideMaxHealth = false)
...
         }
 
+        /// <summary>
+        /// Uses an item for giving its effect.
+        /// </summary>
+        /// <param name="usableItem">The item to be used.</param>
+        public void UseItem(ItemType usableItem)
+        {
+            if (Item.Create(usableItem, this) is not Usable item)
+                throw new Exception($"The provided item [{usableItem}] is not a usable item.");
```
Comment (Ficus-x) [inline Exiled.API/Features/Player.cs:1922]: Maybe throw **new InvalidOperationException**? Also "The provided item [{usableItem}] is not a**n** usalbe item."

**Pull request merged**